Source/WebCore/dom/Document.cpp

```diff
@@ -2989,15 +2989,28 @@ void Document::collectRangeDataFromRegister(Vector<WeakPtr<HighlightRangeData>>&
 {
     for (auto& highlight : highlightRegister.map()) {
         for (auto& rangeData : highlight.value->rangesData()) {
-            // FIXME: For live ranges, we can optimize by only performing this when the range changed.
+            // Skip if positions are not null for static range.
             if (rangeData->startPosition().isNotNull() && rangeData->endPosition().isNotNull() && !rangeData->range().isLiveRange())
                 continue;
+
+            // Skip if a live range did not change start or end.
```

Comment:
Ditto. This comment doesn't add any information beyond what the code already tells us.